From 57c4cb54053ab978f7597eb3bd777782066fba5a Mon Sep 17 00:00:00 2001
From: Tony Liu <Bo.Liu@windriver.com>
Date: Thu, 25 Feb 2010 16:23:42 +0800
Subject: [PATCH] arm_versatile_926ejs: kick off PrimeCell RTC PL031

For RTC PL031, need to explicitly set bit "RTC start" of Control
register(RTCCR) to enable it's counting.

Signed-off-by: Tony Liu <Bo.Liu@windriver.com>
---
 drivers/rtc/rtc-pl031.c |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/rtc/rtc-pl031.c b/drivers/rtc/rtc-pl031.c
index 8900ea7..93839c3 100644
--- a/drivers/rtc/rtc-pl031.c
+++ b/drivers/rtc/rtc-pl031.c
@@ -384,6 +384,9 @@ static int pl031_probe(struct amba_device *adev, const struct amba_id *id)
 		goto out_no_irq;
 	}
 
+	/*Set bit "RTC start" in register RTCCR to enable RTC */
+	writel(0x1, ldata->base + RTC_CR);
+
 	device_init_wakeup(&adev->dev, 1);
 
 	return 0;
-- 
1.7.10.4

